crates/c/src/lib.rs

```diff
@@ -36,6 +37,47 @@ pub extern "C" fn qibo_core_circuit_n_elements(circuit: &Circuit) -> usize {
     circuit.n_elements()
 }
 
+#[no_mangle]
+pub extern "C" fn qibo_core_circuit_n_gates(circuit: &Circuit) -> usize {
+    circuit.n_gates()
+}
+
+#[no_mangle]
+pub extern "C" fn qibo_core_circuit_gate(circuit: &Circuit, gid: usize) -> *const c_char {
+    let gate = circuit.gate(gid);
+    let name = match gate {
+        Gate::One(One::H) => "H",
+        Gate::One(One::X) => "X",
+        Gate::One(One::Y) => "Y",
+        Gate::One(One::Z) => "Z",
+        Gate::Two(Two::CNOT) => "CNOT",
+        _ => todo!()
+    };
+
+    CString::new(name).unwrap().into_raw()
+}
+
+#[no_mangle]
+pub extern "C" fn qibo_core_circuit_elements(circuit: &Circuit, gid: usize, ptr: *mut *const usize, len: *mut usize) {
+    let elements = circuit.elements(gid);
+    // Complaints about what follows are to be directed to ChatGPT
+    let boxed_slice = elements.clone().into_boxed_slice();
+    unsafe {
+        *ptr = boxed_slice.as_ptr();
+        *len = elements.len();
+    }
+    std::mem::forget(boxed_slice);
+}
```

Comment:
Ok, you're avoiding returning in this function, resorting to a more complex solution (with mutable inputs, used to effectively return).

I will fix it, since this should be a very simple function.

Reply:
Actually, I would also prefer to do something here and did not want to avoid returning. I was just not sure how to return a Rust vector to C, so I just used chatGPT (as written in the comment), probably not with the best prompt, to get something working. Same for your next comment (about `_free_elements`), it came together with this.

If you are planning to fix, great! Otherwise I can also have another look. Note, that C does not need to manipulate the vector of `elements`, just read the values. I am not sure if that simplifies it in any way.

Reply:
I'm planning to improve over this.

I'd like to return a structured object (i.e. essentially a smart pointer, like `std::vector`), possibly defined within the `lib.rs` itself.
But I'm not sure if I will be able to access its fields from C (I should try).

Reply:
I agree, and most likely, we should focus on the C++ API rather than the pure C, encapsulating objects in std::array, std::vector and smart pointers. I don't believe we will see interest from collaborators to have a pure C backend soon.

Reply:
Yes, I perfectly agree.

That's why I was also interested in direct usage of `free()` (cf. https://github.com/qiboteam/qibo-core/pull/37#discussion_r1651063990), to limit the overhead in the C API definition, and postpone the proper memory handling to the C++ library.

I will try to have another go at packaging it properly, with a more advanced build tool, since now we'll need the C++ applications (including examples) to depend on the C++ library, that in turn will depend on the C library (whose dependency on `qibo-core` main library is handled by Cargo).